packages/ag-charts-community/src/chart/mapping/themes.ts

```diff
@@ -42,12 +47,72 @@ export const themes: ThemeMap = {
     ...lightThemes,
 };
 
-export function getChartTheme(value?: string | ChartTheme | AgChartTheme): ChartTheme {
+type Unvalidate<T> = { [K in keyof T]?: unknown };
+function validateChartThemeObject(unknownObject: object | null): AgChartTheme | undefined {
+    if (unknownObject === null) {
+        return undefined;
+    }
+
+    let valid = true;
+    const { baseTheme, palette, overrides } = unknownObject as Unvalidate<AgChartTheme>;
+
+    if (baseTheme !== undefined && typeof baseTheme !== 'string' && typeof baseTheme !== 'object') {
```

Comment:
We aren't handling the `typeof baseTheme === 'object'` case 100% correctly here, but I think on balance that is fine since strictly the recursive handling is an undocumented feature.